### ISSUE TITLE
Strange Namespace Behaviors

### ISSUE THREAD

**Issue by kferrone** [fission/fission#2198]
**Describe the bug**
I recently noticed some strange effects with Fission and namespaces. So the chart created these two: fission-functions and fission-builder. 

**To Reproduce**
Create a hello world app with spec dir. By default the namespace should be default. 

1. I use fission spec apply and the yamls in the spec are added to the default namespace and fission creates pods running in the fission-functions  ns. This is what was intended right? 
2. Now I decided to change the ns in all of the yaml files in the spec. I run fission spec apply. Now everything including the actual pods and yamls are all running in the namespace I chose, instead of the pods running in the fission-functions ns.

Seems the fission-functions namespace is only used if the namespace is default. 

I also noticed some inconsistencies in how namespaces were chosen when doing the steps above but only using kubectl. Basically pre-deploy the packages using fission-cli, then apply all the other yaml files using kubectl. 

Oddly enough, no matter what namespace anything ends up in, fission at least handles them all like a boss. So at least I haven't had any actual bugs or hurdles to jump, just seems weird to me. 

**Expected result**
I am pretty sure the intended behavior is to always run the actual workloads, ie pods and deployments, in the `fission-functions` namespace. 
I honestly don't mind having the workloads run in the chosen namespace as well so the pods and fission yamls live together. 
Basically I expect consistency.

**Actual result**
Sometimes the yaml and workload are in the same namespce.
Sometimes the yaml is in the namespace you choose and the workload runs in `fission-functions` namespace, mostly happens when you choose `default` ns.

**Comment by Tri0L** [comment]
I see same issue.
If namespace in yaml == `default`, function pods will be created in `fission-function` namespace.
If another, for example `ns1`, function pods will be created in `ns1` namespace

**Comment by sanketsudake** [comment]
Just capturing slack thread discussion here https://fissionio.slack.com/archives/C3LUX6BBP/p1661328557287589

cc @shubham-bansal96

**Comment by sanketsudake** [comment]
Duplicate of #1482 . Please track #1482 for further updates.
